text/0000-test-directories.md

```diff
@@ -0,0 +1,56 @@
+- Start Date: 2020-01=09
+- Relevant Team(s): Ember.js
+- RFC PR: https://github.com/emberjs/rfcs/pull/575
+- Tracking:
+
+# Test Directories
+
+## Summary
+
+This RFC proposes that we change the names of an Ember app's test directories to reflect the names
+in the guides. For example, instead of `tests/acceptance/*-test.js`, we should have
+`tests/application/*-test.js`.
+
+## Motivation
+
+Renaming test directories to match the Guides will make it easier to have a shared language
+between tests, and to reduce confusion for newer users when using Ember CLI.
+
+## Detailed design
+
+These are the directory transformations that need to happen:
+
+- tests/acceptance -> tests/application
+- tests/integration > tests/rendering
+- tests/unit/utils -> tests/unit
+- tests/unit/controllers -> tests/container/controllers
+- tests/unit/routes -> tests/container/routes
+- tests/unit/services -> tests/container/services
```

Comment:
These list does not include all directories, in which tests are created by existing blueprints. All folders used by Ember Data to place tests in are missing:

- tests/unit/adapters
- tests/unit/models
- tests/unit/serializers
- tests/unit/transforms

Also the tests created for components and helpers if `--unit` flag is parsed are missing:

- tests/unit/components
- tests/unit/helpers

Same it true for not that popular features like mixins and (instance) initializers.

The RFC should also say what should happen to these. Not sure how the RFC should handle folders in `tests` that are used by addons not being part of the default blueprint. I guess they should not be touched but this should be stated explicitly in my opinion.

Reply:
I wasn't attempting to solve for Ember Data unit tests! Those generators can keep doing whatever they want IMO. @runspired or someone else from ED, do you want to expand this RFC to address those as well?